shared-module/displayio/Display.c

```diff
@@ -240,11 +243,23 @@ STATIC bool _refresh_area(displayio_display_obj_t* self, const displayio_area_t*
     if (!displayio_display_core_clip_area(&self->core, area, &clipped)) {
         return true;
     }
-    uint16_t subrectangles = 1;
     uint16_t rows_per_buffer = displayio_area_height(&clipped);
     uint8_t pixels_per_word = (sizeof(uint32_t) * 8) / self->core.colorspace.depth;
     uint16_t pixels_per_buffer = displayio_area_size(&clipped);
-    if (displayio_area_size(&clipped) > buffer_size * pixels_per_word) {
+    
+    uint16_t subrectangles = 1;
+    // for SH1107 and other boundary constrained controllers
+    //      write one single row at a time
+    if (self->SH1107_addressing) {
+        subrectangles = rows_per_buffer;  // vertical (column mode) write each separately (height times)
+    }
+
+    // Skip this recalculation of subrectangles if SH1107 (each subrectangle is a single row)
+    // [mdroberts1243] I am worried/confused about the pixels_in_byte_share_row calculation though
+    //       since it makes sense to be on a byte boundary (actually a page boundary too)
+    //       seems to work as is though.
+    if ((displayio_area_size(&clipped) > buffer_size * pixels_per_word)
+                    && (!self->SH1107_addressing)) {
```

Comment:
I think it may be clearer to add an `if` before this one where SH1107_addressing is true. This condition then goes back to normal as an `else if` on the quirk. Then, set both `subrectangles`, like you do above, and `rows_per_buffer` to 1 so you don't need to set `remaining_rows` to 1 each loop below. This approach also matches the `if` body here which sets both of those variables. I think the logic in the loop will just work as it originally was then. The only thing you'll need to add there is passing the quirk status into setting the region.

Reply:
I've coded this change but will take some time to test  a bit!

Not sure what you mean by "The only thing you'll need to add there is passing the quirk status into setting the region."  The quirk is already passed into setting the region.